### PR TITLE
Add Ecdsa support

### PR DESCRIPTION
**Pull request by danbev**
This commit add support for ECDSA keys.

The motivation for adding this is to be able to use keys that are generated by cosign, and in-toto command line tools are now able to accept ecdsa keys which is also a motivating factor.

Fixes: https://github.com/in-toto/in-toto-rs/issues/43

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

### CODE REVIEW

**Comment by adityasaky** [comment]
This looks good to me but I'm going to wait for another set of eyes to take a look. Thanks, @danbev!

**Comment by danbev** [comment]
@adityasaky Thanks for taking a look and also for merging the other PRs!

File: src/crypto.rs [hunk 1]
```diff
@@ -1675,6 +1687,22 @@ mod test {
         assert_ne!(hasher256.finish(), hasher512.finish());
     }
 
+    #[test]
+    fn test_public_key_from_pem_spki() {
```
Comment (adityasaky) [inline src/crypto.rs:1691]: Can we update this test to be similar to `PublicKey::from_spki`? Test for all key types etc. A bit redundant but I think it'd be nice. WDYT?

Comment (danbev) [reply src/crypto.rs:1691]: That makes sense, I'll take a look at this :+1:

Comment (adityasaky) [reply src/crypto.rs]: Looks good!

**Comment by adityasaky** [comment]
Thanks @danbev!

**Pull request merged**